java/java.lsp.server/vscode/src/panels/SshGuidePanel.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import * as vscode from "vscode";
+import * as path from "path";
+import * as fs from "fs";
+import * as Handlebars from "handlebars";
+import { promisify } from "util";
+import { toggleGuideFor, shouldHideGuideFor, removeGuidePropertiesFromStateFor, NotificationActionType, sshConfigLocation, StatePropertyPrefix, dummyKeyPathLocation } from "./guidesUtil";
+
+const readFile = promisify(fs.readFile);
+
+export class SshGuidePanel {
```

Comment:
The `SshGuidePanel` and `RunImageGuidePanel` are +- the same except for constants. I suspect the Guide concept will continue to expand for more futures - it would be better to have an abstract superclass GuidePanel and either configured instances for each Guide or subclasses.